src/main/java/com/bgsoftware/superiorskyblock/mission/MissionsManagerImpl.java

```diff
@@ -286,6 +286,13 @@ public void rewardMission(Mission<?> mission, SuperiorPlayer superiorPlayer,
                     return;
                 }
 
+                if (checkAutoReward && isAutoReward(mission) && !BuiltinModules.MISSIONS.isAutoRewardOutsideIslands() &&
+                        !plugin.getGrid().isIslandsWorld(superiorPlayer.getWorld())) {
+                    if (result != null)
+                        result.accept(false);
+                    return;
+                }
+
```

Comment:
Cache the result of `isAutoReward` for the mission, as it is checked later as well.
Also, because `checkAutoReward` is also checked twice, do something like this:

```
if (checkAutoReward) {
    boolean shouldAutoReward = isAutoReward(mission);
    if(shouldAutoReward && !BuiltinModules.MISSIONS.isAutoRewardOutsideIslands() && !plugin.getGrid().isIslandsWorld(superiorPlayer.getWorld())) {
        ...
    }
    
    if(!shouldAutoReward) {
        ...
    }
}
```